src/main/java/it/pagopa/transactions/client/PaymentGatewayClient.java

```diff
@@ -422,8 +422,9 @@ private Mono<Tuple2<String, FieldsDto>> requestNpgBuildSession(
                                         );
                                     }
                                     if (statusCode.is4xxClientError()) {
-                                        return new AlreadyProcessedException(
-                                                authorizationData.transactionId()
+                                        return new NpgNotRetryableErrorException(
+                                                "Npg received 4xx error for transactionId: %s, correlationId: %s".formatted(authorizationData.transactionId().value(), correlationId),
```

Comment:
```suggestion
                                                "Npg received 4xx error for transactionId: [%s], correlationId: [%s]".formatted(authorizationData.transactionId().value(), correlationId),
```
Use square brackets here :)